pinot-core/src/main/java/com/linkedin/pinot/core/data/manager/realtime/LLRealtimeSegmentDataManager.java

```diff
@@ -376,15 +380,15 @@ private void processKafkaEvents(Iterable<MessageAndOffset> messagesAndOffsets, L
         transformedRow = _fieldExtractor.transform(decodedRow, transformedRow);
 
         if (transformedRow != null) {
-          _serverMetrics.addMeteredTableValue(_metricKeyName, ServerMeter.REALTIME_ROWS_CONSUMED, 1);
+          realtimeRowsConsumedMeter = _serverMetrics.addMeteredTableValue(_metricKeyName, ServerMeter.REALTIME_ROWS_CONSUMED, 1, realtimeRowsConsumedMeter);
```

Comment:
Does it make sense to batch the metric emission every k rows, instead of each row?